salt/client/ssh/ssh_py_shim.py

```diff
@@ -273,6 +274,11 @@ def main(argv):  # pylint: disable=W0613
     """
     Main program body
     """
+    try:
+        lock_fh = open("{0}.lock".format(OPTIONS.saltdir), "w")
+        fcntl.flock(lock_fh.fileno(), fcntl.LOCK_EX)
+    except (IOError, OSError):
```

Comment:
This is Python3.6+ code only, we only need to care about `OSError`. Since we're just `pass`ing, we could also use the following. Just learned about that recently, we don't have to start using that **now**.

```python
with contextlib.suppress(OSError):
    code_that_might_raise_os_error()
```

Reply:
ssh_py_shim is working on salt ssh system side

Reply:
Right, but Salt 3002.2 is only used by SSH clients that run Python3.6+. Python2 SSH clients don't use the code from this branch.

Reply:
I think we are mixing diffetent things in this discussion.
This exact `ssh_py_shim.py` file is used by salt-api to prepare the script passed in base64 to the shell script by ssh to the managed system, so this code is used for SLE11 also and I tested it on it.

Reply:
`fileno()` call removed and comments appended with https://github.com/openSUSE/salt/pull/414/commits/89bab932fb7307a501b5ba0bee6f959a8c1973ae
Are you ok with the rest part? The code in this part must be Python 2.6 compatible

Reply:
Thanks for the explanation, you're right this file is special in that it is used on all ssh clients, even those that get a different salt codebase as their execution environment.

Reply:
`IOError` and `OSError` should be used both for handling the lock as Python2 producing IOErrror while Python3 - OSError, but in case of calling `os.remove` Python2 and Python3 producing OSError